ApiEverest/CrudApi/Controllers/CostumerController.cs

```diff
@@ -0,0 +1,96 @@
+﻿using AppModels;
+using AppServices.Services;
+using AutoMapper;
+using DomainModels.Entities;
+using DomainServices.Services;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+
+namespace ApiEverest.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CostumerController : ControllerBase
+    {
+        private readonly ICustomerAppService _customerAppService;
+        
+        public CostumerController(ICustomerAppService customerAppService)
+        {
+            _customerAppService = customerAppService ?? throw new ArgumentNullException(nameof(customerAppService));
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            var response = _customerAppService.GetAll();
+           
+            return Ok(response);
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetById(long id)
+        {
+            try
+            {
+                var response = _customerAppService.GetById(id);
+                return Ok(response);
+            }
+            catch (ArgumentNullException exception)
+            {
+                var message = exception.InnerException?.Message ?? exception.Message;
+                return NotFound(message);
+            }
+        }
+
+        [HttpPost]
+        public IActionResult Create([FromBody] CustomerCreateDto customerEntity)
+        {
+            try
+            {
+                _customerAppService.Create(customerEntity);
+                return Created("Id:", _customerAppService);
```

Comment:
```suggestion
                _customerAppService.Create(customerEntity);
                return Created("Id:", _customerAppService);
```
Você está retonando ICustomerAppService não acho que é isso que você quer retornar. Deveria pegar o retorno da linha 51 (o CustomerId) e colocar dentro do Created